ledger/src/blockstore.rs

```diff
@@ -356,14 +359,16 @@ impl Blockstore {
                 .db
                 .delete_range_cf::<cf::Index>(&mut write_batch, from_slot, to_slot)
                 .unwrap_or(false)
-            & self
-                .db
-                .delete_range_cf::<cf::TransactionStatus>(&mut write_batch, from_slot, to_slot)
-                .unwrap_or(false)
             & self
                 .db
                 .delete_range_cf::<cf::Rewards>(&mut write_batch, from_slot, to_slot)
                 .unwrap_or(false);
+        if let Some(index) = self.toggle_transaction_status_index(&mut write_batch, to_slot)? {
+            columns_empty &= &self
```

Comment:
Read-Writes to the transaction statuses from the TransactionStatus thread can interleave  with the `toggle_transaction_status_index -> delete_range_cf` batch here. For instance, if:

1) The transaction_writer thread reads `TransactionStatus` 0 as being not frozen
2) Ledger cleanup thread toggles to frozen here and committed
3)  transaction_writer thread  commits the old `TransactionStatus` 0, making it not frozen again

Reply:
Yeah, I follow. Does it seem viable to wrap the `transaction_status_index_cf` in a lock?

Reply:
Hmm maybe just a `RwLock<active transaction status>` indicating which transaction status column is the active one.

The toggle function only needs to grab it to set and then can release it before commit
The write_function would need the lock for the entire `read->write->commit`